lib/MonadicRewrite.thy

```diff
@@ -178,6 +182,12 @@ lemma monadic_rewrite_named_bindE:
   apply simp
   done
 
+lemma monadic_rewrite_drop_return:
+  "monadic_rewrite F E P f g \<Longrightarrow> monadic_rewrite F E P (do y \<leftarrow> return r; f od) g"
```

Comment:
suggest `return x` or something that hints at a value... `r` is usually a relation/set.

Reply:
also happier with `return x`, but I'd like to point out that a set should be `R` not `r` (a relation could be either).